docs/UI.puml

```diff
@@ -0,0 +1,9 @@
+@startuml
+Class TextUI{
+    + getUserCommand()
+    + showMessage(message: Object)
+}
+Scanner -* TextUI
+PrintStream <-- TextUI
```

Comment:
Should PrintStream be considered as a composition as well if Scanner is? PrintStream and Scanner should not exist if TextUi does not exist is what I'm understanding.

Reply:
I did it like that because i thought the PrintSream Object is System.out. So it should exists even when TextUi is destroyed

Reply:
I may support Yongsheng's point of view. The reason is that all interactive I/O(ui) is done by the ui held by Main, which means the UI cannot be a independent component out of Main? (The System.out is just a API called by PrintSream Object but not PrintSream Object itself, if we wanna make a GUI can directly change the System.out to other output format and our App will still run as expected)

Reply:
Removed the Class Diagram for now instead to see how we can improve on it in the future.